admin/preprint_providers/views.py

```diff
@@ -366,7 +364,7 @@ def share_post(self, preprint_provider):
                     'attributes': {
                         'homePage': preprint_provider.domain if preprint_provider.domain else '{}/preprints/{}/'.format(osf_settings.DOMAIN, preprint_provider._id),
                         'longTitle': debug_prepend + preprint_provider.name,
-                        'iconUrl': '{}{}{}/square_color_no_transparent.png'.format(settings.OSF_URL, osf_settings.PREPRINTS_ASSETS, preprint_provider._id)
```

Comment:
This line was causing issues in auto setting up preprint providers. Basically if there was no icon for the provider it failed to get a share token.
1. Is it safe to say that per your note on the PR that icons are required now, this won't be an issue anymore?
2. Anything enforcing that other than everything fails?
3. Is there an easy way to make sure setting things up locally has an icon without having to add one manually for each?

Reply:
1) It's up to Community to not break this now. It's documented in their "PreprintProvider Creation Instructions" gdoc
2) Everything wouldn't necessarily fail in their absence -- `get_asset_url` defaults to `None` if it can't find one. More validation on this endpoint might be a good thing, though...
3) Designed in such a way that there isn't a need to set icons up locally, unless you want to make sure a local instance of SHARE knows where the icons are.